nestkernel/recording_device.cpp

```diff
@@ -595,12 +642,30 @@ nest::RecordingDevice::record_event( const Event& event, bool endrecord )
   const Time stamp = event.get_stamp();
   const double offset = event.get_offset();
   const double weight = event.get_weight();
+  const long port = event.get_port();
+  const long rport = event.get_rport();
 
-  // std::cout << "recording device sender: " << sender << std::endl;
+  index receiver = -1;
+  if ( P_.withreceivergid_ )
+  {
+    const WeightRecorderEvent* wr_e =
+      dynamic_cast< const WeightRecorderEvent* >( &event );
+    if ( wr_e != 0 )
+    {
+      receiver = wr_e->get_receiver_gid();
+    }
+    else
+    {
+      receiver = event.get_receiver_gid();
+    }
+  }
```

Comment:
What you are suggesting is unfortunately not working, I need a `dynamic_cast` here.

Consider that other recording devices like the `spike_detector` also have the parameter `withtargetgid` now, and this function receives also `SpikeEvents` which does not have the field `receiver_gid`. To get a reasonable output for this case, I implemented `get_receiver_gid()` in `Event` which returns the `gid` of the receiver (unlike the `WeightRecorderEvent` which returns the `gid` of the post-synaptic neuron).

I also can't use a `static_cast` from one derived class to another (i.e. `SpikeEvent` to `WeightRecorderEvent`), as this results in undefined behavior.